managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1921,6 +1927,7 @@ void persistPosition(final LedgerHandle lh, final PositionImpl position, final V
                     position);
         }
 
+        checkNotNull(lh);
```

Comment:
I think we should fail `callback` rather throwing RuntimeException as it is also being called from [non-try-catch-async-callback](https://github.com/yahoo/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java#L1857)